### PR TITLE
Derivatives new

### PR DESCRIPTION
**Pull request by Lakshya18jul**

### CODE REVIEW

File: src/questions/derivatives.yaml [hunk 1]
```diff
@@ -0,0 +1,948 @@
+- uuid: e0b6dff3-946e-4e8d-9f23-8728fbb37c2a
+  type: SingleChoice
+  content: What determines the value of derivatives?
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: What primarily determines the value of derivatives?
```

File: src/questions/derivatives.yaml [hunk 2]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: An asset from which it derives its value
+      key: B
+    - content: Fixed Price set by the owner
+      key: C
+    - content: Standardised Prices are set by the Government.
+      key: D
+- uuid: ad7116fc-e76e-4e5b-b98f-55cea5c7e9e3
+  type: SingleChoice
+  content: Financial Derivatives consist of-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: Which of the following is a derivative? 
```

File: src/questions/derivatives.yaml [hunk 3]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: Bonds
+      key: B
+    - content: Futures
+      key: C
+    - content: None of the above 
+      key: D
+- uuid: 16f39eef-f4d9-47dc-9c9d-2f3b34dea7f9
+  type: SingleChoice
+  content: Porfolio Hedging is done by bank manager to-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Is it "interest rate" risk or "overall" risk ?

File: src/questions/derivatives.yaml [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: Increase reinvestment risk
+      key: B
+    - content: Increase exchange rate risk.
+      key: C
+    - content: Increase the probability of gains.
+      key: D
+- uuid: b59e8a13-3f22-4b14-9f27-c9050ce63a62
+  type: SingleChoice
+  content: Payoffs in financial derivatives are connected to-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: I didn't get this question

File: src/questions/derivatives.yaml [hunk 5]
```diff
@@ -0,0 +1,948 @@
...
+      key: B
+    - content: Previously issued securities.
+      key: C
+    - content: Government standardising the return rates.
+      key: D
+- uuid: da5f00bd-a592-4463-9508-bdf2c85a9741
+  type: SingleChoice
+  content: To hedge a risk for long position we should-
+  hint: NoHint
+  explanation: Hedging risk for long position can be attained by taking a short position
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  explanation: Hedging risk for long position can be attained by taking a opposite position which reduces the exposure/risk. 
```

File: src/questions/derivatives.yaml [hunk 6]
```diff
@@ -0,0 +1,948 @@
...
+    - D
+  subTopics:
+    - "introduction-to-derivatives"
+  difficultyLevel: "Low"
+  choices:
+    - content: Commodity
+      key: A
+    - content: Bond
+      key: B
+    - content: IOU Document
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Add full form of IOU

File: src/questions/derivatives.yaml [hunk 7]
```diff
@@ -0,0 +1,948 @@
...
+      key: C
+    - content: Stock
+      key: D
+- uuid: 84970ae5-4563-4404-9d8b-b59068386f25
+  type: MultipleChoice
+  content: When a bank manager hedges his portfolio by selling futures, he
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: We need to update the question with the exact position information also

File: src/questions/derivatives.yaml [hunk 8]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: removes his loss chances
+      key: B
+    - content: increases his chances of gains
+      key: C
+    - content: All of the above
+      key: D
+- uuid: 3531d44a-67df-4956-98b3-72b99c02daad
+  type: SingleChoice
+  content: What are the advantages to hedge a portfolio?
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Almost same question is above also `Porfolio Hedging is done by bank manager to-`

File: src/questions/derivatives.yaml [hunk 9]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: Minimise the risk chances
+      key: B
+    - content: Profit from capital gains
+      key: C
+    - content: All of the above
+      key: D
+- uuid: e43c62c2-5cf4-4e77-bbb3-8901d01a1263
+  type: SingleChoice
+  content: Which of the following contracts wants the investor to buy assets in future time period?
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: Which of the following contracts would the investor buy if he believes that the assets will increase in future time period?
```

File: src/questions/derivatives.yaml [hunk 10]
```diff
@@ -0,0 +1,948 @@
...
+  content: In a short contracts the investors
+  hint: NoHint
+  explanation: explanation
+  answerKeys:
+    - A
+  subTopics:
+    - "introduction-to-derivatives"
+  difficultyLevel: "Low"
+  choices:
+    - content: Sell securities in future time
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: The investor believes that the asset price will go down in the future. Please update the options to be more relevant

File: src/questions/derivatives.yaml [hunk 11]
```diff
@@ -0,0 +1,948 @@
...
+      key: B
+    - content: Future Hedge
+      key: C
+    - content: None of the above
+      key: D
+- uuid: f3e168ef-a8b3-432c-807b-135fd306335d
+  type: SingleChoice
+  content: What are Speculations in Derivative Market?
+  hint: NoHint
+  explanation: Speculations means using market fluctuations to gain profits in long term.
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Speculator form some beliefs and based on that they take a long or short position. Here Option `A` and `B` are partially correct.

We should update explanation and options for this question

File: src/questions/derivatives.yaml [hunk 12]
```diff
@@ -0,0 +1,948 @@
...
+      key: B
+    - content: Default risk
+      key: C
+    - content: All of the above
+      key: D
+- uuid: ad133f59-6685-42fe-b47f-d2d575140e1c
+  type: SingleChoice
+  content: How are forward contracts better over futures?
+  hint: Benefits of Forward and Futures contract
+  explanation: Forward and Futures contract are identical in temrns of standardisation, default risk and they are equally liquid as well.
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: `explanation: Forward and Futures contract are identical in temrns of standardisation, default risk and they are equally liquid as well.`  is this true? as I thought forward contracts are not standard

File: src/questions/derivatives.yaml [hunk 13]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: They have lower default risk
+      key: B
+    - content: They are more liquid
+      key: C
+    - content: None of the above
+      key: D
+- uuid: f4abb691-9853-425c-b241-f72cfebfa277
+  type: SingleChoice
+  content: Futures are regularly traded on
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: We should ask about the `first exchange`

File: src/questions/derivatives.yaml [hunk 14]
```diff
@@ -0,0 +1,948 @@
...
+      key: A
+    - content: Reduce the loss chances
+      key: B
+    - content: High portfolio earning
+      key: C
+    - content: All of the above
+      key: D
+- uuid: 051125c0-74d8-463e-a23c-79d251a58c5a
+  type: SingleChoice
+  content: What happens to a bank that has perfectly hedged its portfolio during the time when interest rates fall in the futures market?
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: We need to update the question as `interest rate` is not a very contextual term in Futures

**Review by RobinNagpal — APPROVED** [review]

File: src/questions/derivatives.yaml [hunk 15]
```diff
@@ -0,0 +1,948 @@
...
+      key: B
+    - content: Flexibility
+      key: C
+    - content: Both A and B
+      key: D
+- uuid: f5d9ea8c-5621-4780-bdcd-855729a100ea
+  type: SingleChoice
+  content: Parties who have bought a futures contract and thereby decided to ..... the bonds are said to have taken ....... position.?
+  hint: Buyers and their position in futures market 
+  explanation: Buyers in the futures market when agree to buy the contract are said to take a long position.
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml:349]: Buyers can buy a short position also

**Review by RobinNagpal — CHANGES_REQUESTED** [review]

File: src/questions/derivatives.yaml (hunk 1 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done the changes

File: src/questions/derivatives.yaml (hunk 2 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 3 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Amongst the given options I want the user to chose the best one

File: src/questions/derivatives.yaml (hunk 4 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: In this question I am trying to ask from the user that the financial derivatives are payed off in which asset

File: src/questions/derivatives.yaml (hunk 5 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 6 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 9 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 10 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 11 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done the updation

File: src/questions/derivatives.yaml (hunk 12 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Made the changes

File: src/questions/derivatives.yaml (hunk 13 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 14 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 15 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml:349]: Question statement improved

File: src/questions/derivatives.yaml (hunk 7 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 8 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml [hunk 16]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Bonds
+      key: B
+    - content: Futures
+      key: C
+    - content: None of the above 
+      key: D
+- uuid: 16f39eef-f4d9-47dc-9c9d-2f3b34dea7f9
+  type: SingleChoice
+  content: Portfolio Hedging is done by bank manager to-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml:41]: This question needs to be updated as we didn't read about `Reduce interest rate risk.`

All options need to be updated

File: src/questions/derivatives.yaml (hunk 4 above)
Comment (RobinNagpal) [reply src/questions/derivatives.yaml]: We can just take this off

File: src/questions/derivatives.yaml [hunk 17]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Bond
+      key: B
+    - content: IOU (Investor Owned Utilities) Document 
+      key: C
+    - content: Stock
+      key: D
+- uuid: 84970ae5-4563-4404-9d8b-b59068386f25
+  type: MultipleChoice
+  content: When a bank manager hedges his portfolio in short position by selling futures, he
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Question needs to be replaced

File: src/questions/derivatives.yaml [hunk 18]
```diff
@@ -0,0 +1,951 @@
...
+  explanation: explanation
+  answerKeys:
+    - A
+  subTopics:
+    - "introduction-to-derivatives"
+  difficultyLevel: "Low"
+  choices:
+    - content: Predicts to sell securities in future time when prices go down
+      key: A
+    - content: Predicts to buy securities in future time when prices go up
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Sell now and buy later

File: src/questions/derivatives.yaml [hunk 19]
```diff
@@ -0,0 +1,951 @@
...
+  hint: Hedging reduces the risk
+  explanation: Hedging in futures market reduces the chances of gain and losses.
+  answerKeys:
+    - A
+    - B
+  subTopics:
+    - "types-of-derivatives"
+  difficultyLevel: "Medium"
+  choices:
+    - content: Reduces the gain chnaces
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
    - content: Reduces the gains
```

File: src/questions/derivatives.yaml [hunk 20]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Reduce the loss chances
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
    - content: Reduce the loss
```

File: src/questions/derivatives.yaml [hunk 21]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: buy,long
+      key: B
+    - content: sell,short
+      key: C
+    - content: sell,long
+      key: D
+- uuid: 8d57efe3-c473-4357-82de-7223db6cdb99
+  type: SingleChoice
+  content: When a futures contract expires the price of the contract-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: When a futures contract expires then -
```

File: src/questions/derivatives.yaml [hunk 22]
```diff
@@ -0,0 +1,951 @@
...
+  hint: Futures contract have a fixed price
+  explanation: When a futures contract expires its price equals to price of contract that was agreed upon by the buyer and seller at the time of making of contract.
+  answerKeys:
+    - A
+  subTopics:
+    - "types-of-derivatives"
+  difficultyLevel: "Low"
+  choices:
+    - content: Equals to purchase price of contract
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
    - content: The asset is bought or sold at the price mentioned in the contract
```

Comment (RobinNagpal) [reply src/questions/derivatives.yaml]: Improve other option

File: src/questions/derivatives.yaml [hunk 23]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Equals to mean price over the life of contract
+      key: B
+    - content: Equals to the price of the underlying asset
+      key: C
+    - content: None of the above
+      key: D
+- uuid: 7e890197-fe94-4405-aadf-efcd102f94ff
+  type: SingleChoice
+  content: Removing any riskless profit opportunities in the futures market is known as-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: Execution(or some other synonym) of any riskless profit opportunities in the futures market is known as-
```

File: src/questions/derivatives.yaml [hunk 24]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Micro Hedge
+      key: B
+    - content: Cross Hedge
+      key: C
+    - content: Futures Hedge
+      key: D
+- uuid: 490d9314-dc38-4d25-900e-36d1c52816fe
+  type: SingleChoice
+  content: The amount of outstanding futures contracts is known as-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: The number of outstanding futures contracts is known as- 
```

File: src/questions/derivatives.yaml [hunk 25]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Obligation to trade deriving asset
+      key: B
+    - content: ability to minimise the transaction risk
+      key: C
+    - content: None of the above
+      key: D
+- uuid: f7b24b32-b7ef-4f2e-8b34-992d6690582c
+  type: SingleChoice
+  content: An option that allows the owner to sell an asset at a future time is a
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: An option that allows the owner to sell now and buy an asset later at a future time is a
```

File: src/questions/derivatives.yaml [hunk 26]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Futures options
+      key: B
+    - content: American options
+      key: C
+    - content: Individual options
+      key: D
+- uuid: 8bf31665-8afd-46de-af73-3d4755f90069
+  type: SingleChoice
+  content: Options that can only be executed on maturity is called-
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]:
```suggestion
  content: Type of options that can only be executed on maturity is called-
```

File: src/questions/derivatives.yaml [hunk 27]
```diff
@@ -0,0 +1,951 @@
...
+      key: A
+    - content: Futures Options
+      key: B
+    - content: Stock Options
+      key: C
+    - content: Individual Options
+      key: D
+- uuid: 92fb85a6-f95d-4c30-ad38-01ca1bf64287
+  type: SingleChoice
+  content: Which of the following options gives the owner the right to buy a financial instrument at executed price within a specific duration of time?
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml:689]: Please confirm this. (For both Robin and Lakshya)

File: src/questions/derivatives.yaml [hunk 28]
```diff
@@ -0,0 +1,951 @@
...
+  choices:
+    - content: SNT token
+      key: A
+    - content: STX token
+      key: B
+    - content: SNX token
+      key: C
+    - content: SYX token
+      key: D
+- uuid: 98dc4273-ede6-4a04-8a63-1adb4bf07436
```
Comment (RobinNagpal) [inline src/questions/derivatives.yaml]: Remove this

File: src/questions/derivatives.yaml (hunk 16 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml:41]: Done

File: src/questions/derivatives.yaml (hunk 17 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 18 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 19 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 20 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 21 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 22 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 23 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 24 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 25 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 26 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

File: src/questions/derivatives.yaml (hunk 27 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml:689]: Robin, I have confirmed this is correct. I can explain in the next meet

File: src/questions/derivatives.yaml (hunk 28 above)
Comment (Lakshya18jul) [reply src/questions/derivatives.yaml]: Done

**Pull request merged**